zerver/views/message_fetch.py

```diff
@@ -178,6 +183,10 @@ def by_has(self, query: Query, operand: str, maybe_negate: ConditionTransform) -
         return query.where(maybe_negate(cond))
 
     def by_in(self, query: Query, operand: str, maybe_negate: ConditionTransform) -> Query:
+        # This operator does not support is_web_public_query.
+        assert not self.is_web_public_query
+        assert self.user_profile is not None
```

Comment:
Would it be cleaner to refactor to make this pattern a decorator?

Reply:
Yeah, that'd be a potentially nice follow-up.